### PR TITLE
MINOR: Minor update patch versions

### PR DESCRIPTION
**Pull request by jolshan**
We were missing the versions for these patch releases. The versions are in versions.py and Dockerfile.

I've also added a step to the release guide to update these files. https://cwiki.apache.org/confluence/display/KAFKA/Release+Process
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Review by mjsax — APPROVED** [review]

**Comment by jolshan** [comment]
@satishd or @showuon -- did we notice these versions causing issues for the system tests for the 3.6 release?

**Comment by satishd** [comment]
@jolshan:  @showuon helped in running the system tests for 3.6.0 release. I do not think there were any issues reported during those runs related to these versions.

**Comment by jolshan** [comment]
Running the system tests on this branch led to an error:
```
22:52:44     worker2: + chmod a+rw /opt/kafka-2.6.3
22:52:44     worker2: chmod: cannot access ���/opt/kafka-2.6.3���: No such file or directory
```
I will invesigate this. 2.6.3 seems to be a valid kafka release.

**Comment by jolshan** [comment]
@satishd I see that in the results we didn't try to run for these patch releases in the tests, so I see why they didn't fail. I wonder if those tests were not enabled to be run at the time of the 3.6 release.
https://drive.google.com/drive/folders/1qwIKg-B4CBrswUeo5fBRv65KWpDsGUiS

File: vagrant/base.sh [hunk 1]
```diff
@@ -155,7 +155,7 @@ chmod a+rw /opt/kafka-3.1.2
 get_kafka 3.2.3 2.12
 chmod a+rw /opt/kafka-3.2.3
 get_kafka 3.3.1 2.12
-chmod a+rw /opt/kafka-3.3.1
+chmod a+rw /opt/kafka-3.3.2
```
Comment (jolshan) [inline vagrant/base.sh]: looks we are also missing the get call 🤦‍♀️ I will fix

**Comment by jolshan** [comment]
Reran and the majority of the 2.6.3 and 2.7.2 tests passed. The ones that didn't had a different issue.

**Review by hachikuji — APPROVED** [review]

**Comment by jolshan** [comment]
I will cherrypick to 3.7

**Pull request merged**